core/gql_queries.py

```diff
@@ -45,7 +45,7 @@ def get_queryset(cls, queryset, info):
         elif info.context.user.is_superuser:
             return queryset
         else:
-            queryset = queryset.filter(*filter_validity())
+            pass
```

Comment:
did you have that *filter_validity() set elsewhere ? if not please add a case where the historical is wished but for other cases *filter_validity() must still be used  (in info ?)

Reply:
Yes. It was in schema.py 
`def resolve_role(self, info, **kwargs):
        if not info.context.user.has_perms(CoreConfig.gql_query_roles_perms):
            raise PermissionError("Unauthorized")
        filters = []
        query = Role.objects

        client_mutation_id = kwargs.get("client_mutation_id", None)
        if client_mutation_id:
            filters.append(Q(mutations__mutation__client_mutation_id=client_mutation_id))

        show_history = kwargs.get('show_history', False)
        if not show_history and not kwargs.get('uuid', None):
            filters += filter_validity(**kwargs)

        is_system_role = kwargs.get('is_system', None)
        # check if we can use default filter validity
        if is_system_role is not None:
            if is_system_role:
                query = query.filter(
                    is_system__gte=1
                )
            else:
                query = query.filter(
                    is_system=0
                )
        return gql_optimizer.query(query.filter(*filters), info)`

Reply:
it is exactly here 
        ```
        show_history = kwargs.get('show_history', False)
        if not show_history and not kwargs.get('uuid', None):
            filters += filter_validity(**kwargs)
```

Reply:
Just to be sure, can you confirm that this query is not called else where. I fear that user will get all their role even the removed ones

Reply:
Hello @edarchis, do you think this change will have impact on the way django is loading the users rights ?

Reply:
@delcroip I've changed my idea - in that commit you can take a look into it https://github.com/openimis/openimis-be-core_py/pull/53/commits/9a99fd97c5344117ad9ac9a589fa7aef56c6ce83